### PR TITLE
Create Xen Config Subclass

### PR DESCRIPTION
**Pull request by jirihnidek**
* Implemented subclass of VirtConfigSection for Xen virt. backend
* Implemented custom _validate_server() method for xen
* Fixed one bug in virt.py, when virt type unsupported is set in
  config file
* Fixed all unit tests in test_xen.py

### CODE REVIEW

**Comment by jirihnidek** [comment]
[[8] [CONFIG_EPIC] Create Xen Config Subclass](https://trello.com/c/UfTXRH84/582-8-configepic-create-xen-config-subclass)

**Comment by jirihnidek** [comment]
Please wait with review. I want to test this change with some Xen server.

**Comment by jirihnidek** [comment]
I can confirm that this branch is able to get host-to-guest mapping from xen server.

File: virtwho/virt/xen/xen.py [hunk 1]
```diff
@@ -1,15 +1,66 @@
...
 import requests
 
 from virtwho import virt
 from virtwho.util import RequestsXmlrpcTransport
+from virtwho.config import VirtConfigSection
+
+
+class XenConfigSection(VirtConfigSection):
+    """
+    This class is used for validation of fake virtualization backend
```
Comment (cnsnyder) [inline virtwho/virt/xen/xen.py]: 'fake' -> 'Xen'

File: virtwho/virt/xen/xen.py [hunk 2]
```diff
@@ -1,15 +1,66 @@
...
+    section(s). It tries to validate options and combination of options that
+    are specific for this virtualization backend. In specific, it attempts to read
+    the given file and produces error messages if it is not usable.
+    """
+    VIRT_TYPE = 'xen'
+
+    def __init__(self, section_name, wrapper, *args, **kwargs):
+        super(XenConfigSection, self).__init__(section_name, wrapper, *args, **kwargs)
+
```
Comment (cnsnyder) [inline virtwho/virt/xen/xen.py]: Similar feedback as to the RHEVM PR #105 :

Please add something similar to the following as it seems 'server', 'username', and 'password' are required:

```
 self.add_key('server', validation_method=self._validate_server, required=True)
 self.add_key('username', validation_method=self._validate_username, required=True)
 self.add_key('password', validation_method=self._validate_unencrypted_password, required=True)
```

File: virtwho/virt/xen/xen.py [hunk 3]
```diff
@@ -121,14 +168,14 @@ def getHostGuestMapping(self):
             if version:
                 facts[virt.Hypervisor.HYPERVISOR_VERSION_FACT] = version
 
-            if self.config.hypervisor_id == 'uuid':
+            if self.config['hypervisor_id'] == 'uuid':
```
Comment (cnsnyder) [inline virtwho/virt/xen/xen.py:174]: Should this be moved to a _validate_hypervisor_id method?

Would return an error message if the value provided was not one of those expected and valid for Xen.

**Review by cnsnyder — CHANGES_REQUESTED** [review]
A few requested updates similar to those on the RHEVM PR.

Please take a look and let me know what you think!

Cheers,
Chris

File: virtwho/virt/xen/xen.py (hunk 3 above)
Comment (jirihnidek) [reply virtwho/virt/xen/xen.py:174]: You are right. The `VirtConfigSection` has only lambda function returning `None` in all cases. I will fix it for other virt. backends (hyper-v and libvirtd) in this PR or I can create special PR when Xen and RHEVM fill be merged to config_refactor.

File: virtwho/virt/xen/xen.py (hunk 2 above)
Comment (jirihnidek) [reply virtwho/virt/xen/xen.py]: True. I will fix this for hyper-v and libvirtd too and I will add some unit tests for this case.

File: virtwho/virt/xen/xen.py (hunk 3 above)
Comment (jirihnidek) [reply virtwho/virt/xen/xen.py:174]: Solved in PR #105. When both PRs will be merged to config_refactor branch, then it should work as expected.

**Review by cnsnyder — APPROVED** [review]
Looks good to me!
Waiting to merge on #105
(Please squash before merge!)

**Pull request merged**